### PR TITLE
Reformat configuration

### PR DESCRIPTION
**Pull request by sujeebant**

### CODE REVIEW

File: test/masq.test.js [hunk 1]
```diff
@@ -7,13 +7,18 @@ window.crypto = require('@trust/webcrypto')
 
 const Masq = require('../src')
 const MasqAppMock = require('./mockMasqApp')
-const config = require('../config/config')
+// const config = require('../config/config.prod.json')
```
Comment (leventdem) [inline test/masq.test.js]: commented ?

Comment (sujeebant) [reply test/masq.test.js]: :+1:

File: config/config.prod.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
-  "MASQ_APP_BASE_URL": "https://qwantresearch.github.io/masq-app/",
-  "HUB_URLS": [ "wss://signalhub-jvunerwwrg.now.sh" ],
+  "MASQ_APP_BASE_URL": "should_be_overridden",
```
Comment (G-Ray) [inline config/config.prod.json]: why ?

File: src/index.js [hunk 3]
```diff
@@ -19,19 +19,12 @@ const debug = (function () {
...
-  })
-}
-
 class Masq {
   /**
    * constructor
    * @param {string} appName - The application name
    */
-  constructor (appName, appDescription, appImageURL, signalhubUrl, baseMasqAppUrl) {
+  constructor (appName, appDescription, appImageURL, _options = {}) {
```
Comment (G-Ray) [inline src/index.js]: you should not prefix wiht an underscore

File: test/masq.test.js [hunk 4]
```diff
@@ -7,7 +7,11 @@ window.crypto = require('@trust/webcrypto')
 
 const Masq = require('../src')
 const MasqAppMock = require('./mockMasqApp')
-const config = require('../config/config')
+const localhostConfig = require('../config/config.localhost.json')
```
Comment (G-Ray) [inline test/masq.test.js]: Don't you prevers config.test.json instead ?

File: config/config.prod.json (hunk 2 above)
Comment (sujeebant) [reply config/config.prod.json]: Because we should provide the production url once we have it and override until then

File: test/masq.test.js [hunk 5]
```diff
@@ -7,7 +7,11 @@ window.crypto = require('@trust/webcrypto')
...
+const testOptions = {
+  hubUrls: localhostConfig.HUB_URLS,
```
Comment (G-Ray) [inline test/masq.test.js]: you should edit the json file define keys you expect here: `hubUrls` and `masqAppBaseUrl`

File: src/index.js [hunk 6]
```diff
@@ -8,7 +8,7 @@ const ERRORS = common.errors.ERRORS
 const MasqError = common.errors.MasqError
 const CURRENT_USER_INFO_STR = 'currentUserInfo'
 
-const config = require('../config/config')
+const _config = require('../config/config.prod.json')
```
Comment (G-Ray) [inline src/index.js]: why prefix with `_` ?

File: src/index.js [hunk 7]
```diff
@@ -49,13 +42,20 @@ class Masq {
     this.loginKey = null
     this.loginUrl = null
 
-    if (signalhubUrl) {
-      config.HUB_URLS = [signalhubUrl]
+    // override config with constructor options
+    this.config = {
+      ..._config,
+      HUB_URLS: _options.hubUrls ? _options.hubUrls : _config.HUB_URLS,
```
Comment (G-Ray) [inline src/index.js]: I think we should use lowercase properties

File: src/index.js (hunk 3 above)
Comment (sujeebant) [reply src/index.js]: changed

File: test/masq.test.js (hunk 4 above)
Comment (sujeebant) [reply test/masq.test.js]: changed localhost to test

File: config/config.prod.json (hunk 2 above)
Comment (G-Ray) [reply config/config.prod.json]: ah yes it's true we don't have a production url yet ;)

File: test/masq.test.js (hunk 5 above)
Comment (sujeebant) [reply test/masq.test.js]: changed to have the same key names in Masq and config

File: src/index.js (hunk 6 above)
Comment (sujeebant) [reply src/index.js]: changed _config to jsonConfig

File: src/index.js (hunk 7 above)
Comment (sujeebant) [reply src/index.js]: changed to use the same key names in Masq and config

**Pull request merged**